### PR TITLE
Reconcile S3Client::putObject and MultipartUploader args

### PR DESCRIPTION
**Pull request by jeskew**
Should address #712.

### CODE REVIEW

File: src/S3/MultipartUploader.php [hunk 1]
```diff
@@ -95,7 +95,7 @@ public static function getStateFromService(
      */
     public function __construct(S3Client $client, $source, array $config = [])
     {
-        parent::__construct($client, $source, $config + [
+        parent::__construct($client, $source, array_change_key_case($config) + [
```
Comment (jeremeamia) [inline src/S3/MultipartUploader.php]: I don't use that function very often, but when you need it, it's just what the doctor ordered.

**Comment by jeremeamia** [comment]
:ship:

File: docs/service/s3-multipart-upload.rst [hunk 2]
```diff
@@ -44,6 +44,33 @@ uploader keeps track of it and continues to upload later parts until the entire
...
+        'before_upload' => function (\Aws\Command $command) {
+           // $command is an UploadPart operation
+           $command['RequestPayer'] = 'requester';
+        },
+        'before_complete' => function (\Aws\Command $command) {
+           // $command is a CompleteMultipartUpload operation
+           $command['RequestPayer'] = 'requester';
+        },
+    ]);
+
```
Comment (mtdowling) [inline docs/service/s3-multipart-upload.rst]: Nitpick: I think we use two spaces between sections.

**Comment by mtdowling** [comment]
:ship:

**Pull request merged**